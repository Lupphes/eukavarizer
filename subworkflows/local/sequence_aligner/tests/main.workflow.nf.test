nextflow_workflow {

    name "Test Workflow SEQUENCE_ALIGNER"
    script "../main.nf"
    workflow "SEQUENCE_ALIGNER"
    config "subworkflows/local/sequence_aligner/tests/nextflow.config"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "sequence_aligner"

    test("BWA alignment - Illumina paired-end - stub") {
        options '-stub'

        when {
            params {
                minimap2_flag = false
                bwamem2 = false
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id: 'test', patient: 'patient1', sample: 'sample1', sex: 'XX', status: 0, platform: 'illumina', single_end: false, size: 1, num_lanes: 1],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = Channel.value([[id: 'genome'], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.bam_mapped != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("BWAMEM2 alignment - Illumina paired-end - stub") {
        options '-stub'

        when {
            params {
                minimap2_flag = false
                bwamem2 = true
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id: 'test', patient: 'patient1', sample: 'sample1', sex: 'XX', status: 0, platform: 'illumina', single_end: false, size: 1, num_lanes: 1],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = Channel.value([[id: 'genome'], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.bam_mapped != null }
            )
        }
    }

    test("MINIMAP2 alignment - ONT long reads - stub") {
        options '-stub'

        when {
            params {
                minimap2_flag = true
                bwamem2 = false
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id: 'test_ont', patient: 'patient1', sample: 'sample1', sex: 'XX', status: 0, platform: 'ont', single_end: true, size: 1, num_lanes: 1],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/fastq/test.fastq.gz', checkIfExists: true)
                ])
                input[1] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = Channel.value([[id: 'genome'], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.bam_mapped != null }
            )
        }
    }

    test("Multiple lanes - same sample - stub") {
        options '-stub'

        when {
            params {
                minimap2_flag = false
                bwamem2 = false
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [id: 'test_L001', patient: 'patient1', sample: 'sample1', sex: 'XX', status: 0, platform: 'illumina', single_end: false, size: 1, num_lanes: 2],
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                        ]
                    ],
                    [
                        [id: 'test_L002', patient: 'patient1', sample: 'sample1', sex: 'XX', status: 0, platform: 'illumina', single_end: false, size: 1, num_lanes: 2],
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = Channel.value([[id: 'genome'], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.bam_mapped != null }
            )
        }
    }

    test("Multiple samples - stub") {
        options '-stub'

        when {
            params {
                minimap2_flag = false
                bwamem2 = false
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [id: 'sample1', patient: 'patient1', sample: 'sample1', sex: 'XX', status: 0, platform: 'illumina', single_end: false, size: 1, num_lanes: 1],
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                        ]
                    ],
                    [
                        [id: 'sample2', patient: 'patient1', sample: 'sample2', sex: 'XX', status: 0, platform: 'illumina', single_end: false, size: 1, num_lanes: 1],
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = Channel.value([[id: 'genome'], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.bam_mapped != null }
            )
        }
    }
}
