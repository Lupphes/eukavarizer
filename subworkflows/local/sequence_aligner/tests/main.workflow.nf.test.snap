{
    "BWA alignment - Illumina paired-end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "patient": "patient1",
                                "sample": "sample1",
                                "sex": "XX",
                                "status": 0,
                                "platform": "illumina",
                                "single_end": false,
                                "n_fastq": 1,
                                "data_type": "bam",
                                "id": "sample1-illumina-false"
                            }
                        },
                        [
                            "sample1.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "bam_mapped": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "patient": "patient1",
                                "sample": "sample1",
                                "sex": "XX",
                                "status": 0,
                                "platform": "illumina",
                                "single_end": false,
                                "n_fastq": 1,
                                "data_type": "bam",
                                "id": "sample1-illumina-false"
                            }
                        },
                        [
                            "sample1.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T23:17:11.311916904"
    }
}